src/components/MTableHeader/index.js

```diff
@@ -10,52 +10,101 @@ import { Tooltip, withStyles } from '@material-ui/core';
 import * as CommonValues from '../../utils/common-values';
 
 export function MTableHeader({ onColumnResized, ...props }) {
-  const [
-    { resizingColumnDef, lastX, lastAdditionalWidth }, // Extract the props to use instead of the whole state object
-    setState
-  ] = React.useState({
-    lastX: 0,
-    resizingColumnDef: undefined
-  });
+  const [resizing, setResizing] = React.useState(undefined);
+  const [lastX, setLastX] = React.useState(0);
 
   const handleMouseDown = (e, columnDef) => {
-    let target = e.clientX;
-    setState((prevState) => ({
-      ...prevState,
-      lastAdditionalWidth: columnDef.tableData.additionalWidth,
-      lastX: target,
-      resizingColumnDef: columnDef
-    }));
+    const target = e.clientX;
+    const th = e.target.closest('th');
+    const currentWidth =
+      th && Math.round(+window.getComputedStyle(th).width.slice(0, -2));
+    let initialColWidths = resizing?.initialColWidths;
+    let nextWidth;
+    let nextCol;
+    if (props.tableWidth === 'full') {
+      const nextTh = th.nextSibling;
+      nextWidth =
+        nextTh &&
+        Math.round(+window.getComputedStyle(nextTh).width.slice(0, -2));
+      nextCol = props.columns.find(
+        (c) => c.tableData.id === columnDef.tableData.id + 1
+      );
+    } else if (!initialColWidths) {
+      // Ensure we have all column widths in pixels
+      initialColWidths = Array.from(th.parentNode.children).map((th) =>
+        Math.round(+window.getComputedStyle(th).width.slice(0, -2))
+      );
+    }
+
+    setLastX(target);
+    setResizing({
+      col: columnDef,
+      nextCol,
+      lastColData: { ...columnDef.tableData, width: currentWidth },
+      ...(nextCol && {
+        lastNextColData: { ...nextCol.tableData, width: nextWidth }
+      }),
+      initialColWidths
+    });
   };
+
+  const constrainedColumnResize = (col, lastWidth, offset) => {
+    // Extra max/min are to avoid sudden column changes when a column that starts without
+    // an explicit width is resized
+    const constrainedNewWidth = Math.min(
+      Math.max(col.maxWidth || 10000, lastWidth), // Avoid sudden decrease in column width
```

Comment:
Could we move those to constants outside of the component to reduce the amount of magic numbers and make it easier to extract them to props later?

Reply:
I wasn't happy with that either!
Moved to constants - I left in it, as that felt step closer to adding as prop later.